include/eve/algo/container/soa_vector2.hpp

```diff
@@ -0,0 +1,371 @@
+    soa_vector(std::size_t n, value_type v, Allocator a = Allocator{})
+              : soa_vector(no_init,n,a)
+    {
+      fill(v);
+    }
+
+    //! Constructs the container with the contents of the initializer list `l`.
+    soa_vector( std::initializer_list<Type> l, Allocator a = Allocator{})
+              : storage(a, l.size()), size_{l.size()}
+    {
+      auto ptr = l.begin();
+      for(std::size_t i = 0;i < size(); ++i)
+      {
+        eve::write(*ptr++, begin()+i);
+      }
+    }
+
+    //==============================================================================================
+    //! @}
+    //==============================================================================================
+
+    //==============================================================================================
+    //! @name Capacity
+    //! @{
+    //==============================================================================================
+    //! Returns the number of elements in the container
+    EVE_FORCEINLINE std::size_t size()  const noexcept { return size_; }
+
+    //! Returns the number of elements that the container has currently allocated space for
+    EVE_FORCEINLINE std::size_t capacity()  const noexcept { return storage.capacity_; }
+
+    //! Requests the removal of unused capacity.
+    EVE_FORCEINLINE void shrink_to_fit() { grow(size(), size()); }
+
+    //! Checks if the container has no elements
+    EVE_FORCEINLINE bool empty() const noexcept { return !size(); }
+
+    //! @brief Increase the capacity of the vector to a value that's greater or equal to `n`.
+    //! If `n` is greater than the current capacity(), new storage is allocated and all iterators,
+    //! including the past-the-end iterator, and all references to the elements are invalidated.
+    //! Otherwise, the method does nothing.
+    //! @warning reserve() does not change the size of the vector.
+    //! @param n New capacity of the `eve::soa_vector`
+    void reserve(std::size_t n)
+    {
+      if(n > capacity())
+      {
+        // Grow capacity to n
+        auto const sz = size_;
+        grow(n, size());
+        size_ = sz;
+     }
+    }
+
+    //==============================================================================================
+    //! @}
+    //==============================================================================================
+
+    //==============================================================================================
+    //! @name Modifiers
+    //! @{
+    //==============================================================================================
+    //! @brief Clear the contents of the container.
+    void clear() { size_ = 0; }
+
+    //! @brief Removes an element from the container.
+    //! Has the same invalidation semantics as std::vector.
+    //! end() iterator is not a valid pos.
+    iterator erase(const_iterator pos) { return erase(pos, pos+1); }
```

Comment:
This is overhead. 

How about we extract `eve::algo::copy_backward` into a function and call that from both `erase`.

Reply:
This pos + 1. Is overhead :(
And then distance back...

I don't know - we'll look at asm and make decisions.

But like, we expect this funciton to be inlined, right?
And then `erase(f, l)` to be not inlined?

That means for 3 fields you do + 3 pointer additions, put them all in a struct, pusht that struct on the stack.

How about an internal erase with 2 indexes? Subtracting zip iterators is the same as subtracting the first pointers only.